### ISSUE TITLE
unable to shut down server

### ISSUE THREAD

**Issue by shanedsnyder** [mochi-hpc/mobject-test#32]
In GitLab by @roblatham00 on Jan 15, 2020, 15:59

If one starts a mobject server, then uses the command line tool to shut it down, the server only partially shuts down: the server will no longer respond to RPCs, but it is still running waiting for... something.

Steps to reproduce:

```
% ./src/server/mobject-server-daemon sockets: mobject.cfg &
% ./src/server/mobject-server-ctl mobject.cfg shutdown
# no errors visible to client, but server is still running. 
# try to shut it down again:
[0] hodor:~/work/mobject-store/build (master *=)% ./src/server/mobject-server-ctl mobject.cfg shutdown
margo_forward_timed(handle, &observe_req, SSG_DEFAULT_OP_TIMEOUT): HG_TIMEOUT
unable to send observe request (ret: -1; 17623649815270081923 (nil) (nil))
Error: Unable to observe SSG server group
```

**Comment by shanedsnyder** [comment]
In GitLab by @shanedsnyder on Jan 16, 2020, 15:29

Fixed with 83ee12cd -- Mobject needs to use Margo pre-finalize callbacks for shutting down SSG since it relies on an active Margo progress thread to complete.

**Comment by shanedsnyder** [comment]
In GitLab by @shanedsnyder on Jan 16, 2020, 15:29

closed
